articles/custom_storage_location.md

```diff
@@ -137,164 +150,234 @@ For read-write permissions (you allow Synapse to upload and retrieve files)
+{% tabs %}
 
-and save the file as **owner.txt**
+{% tab Python %}
+{% highlight python %}
+# Set storage location
+import synapseclient
+import json
+syn = synapseclient.login()
+PROJECT = 'syn12345'
 
-<img src="/assets/images/save-as-owner.jpg">
+destination = {'uploadType':'S3', 
+               'concreteType':'org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting',
+               'bucket':'nameofyourbucket'}
+destination = syn.restPOST('/storageLocation', body=json.dumps(destination))
 
-click on the new bucket name to open up your new bucket
+project_destination ={'concreteType': 'org.sagebionetworks.repo.model.project.UploadDestinationListSetting', 
+                      'settingsType': 'upload'}
+project_destination['locations'] = [destination['storageLocationId']]
+project_destination['projectId'] = PROJECT
 
-<img src="/assets/images/select-bucket.jpg">
+project_destination = syn.restPOST('/projectSettings', body = json.dumps(project_destination))
+{% endhighlight %}
+{% endtab %}
 
-and click the **Upload** button
+{% tab R %}
+{% highlight r %}
+#set storage location
+library(synapseClient)
+synapseLogin()
+projectId <- 'syn12345'
 
-<img src="/assets/images/click-upload.jpg">
+destination <- list(uploadType='S3', 
+                    concreteType='org.sagebionetworks.repo.model.project.ExternalS3StorageLocationSetting',
+                    bucket='nameofyourbucket')
+destination <- synRestPOST('/storageLocation', body=destination)
 
-to show the upload dialog
+projectDestination <- list(concreteType='org.sagebionetworks.repo.model.project.UploadDestinationListSetting', 
+                           settingsType='upload')
+projectDestination$locations <- list(destination$storageLocationId)
+projectDestination$projectId <- projectId
 
-<img src="/assets/images/upload-dialog.jpg">
+projectDestination <- synRestPOST('/projectSettings', body = projectDestination)
+{% endhighlight %}
+{% endtab %}
 
-click on **Add Files**
+{% tab Web %}
+ Navigate to your **Project/Folder -> Tools -> Change Storage Location**. In the resulting pop-up, select the `Amazon S3 Bucket` option and fill in the relevant information, where Bucket is the name of your external bucket, Base Key is the name of the folder in your bucket to upload to, and Banner is a short description such as who owns the storage location:
 
-<img src="/assets/images/click-add-file.jpg">
+<img id="image" src="/assets/images/external_s3.png">
 
-and select the **owner.txt** file you created earlier
+{% endtab %}
 
-<img src="/assets/images/select-owner-file.jpg">
+{% endtabs %}
 
-click on **Start Upload**
 
-<img src="/assets/images/click-start-upload.jpg">
+<br/>
 
-and wait for the upload to finish
+### Example of uploading a file to S3 bucket
 
-<img src="/assets/images/wait-until-done.jpg">
+A `FileHandle` is a Synapse representation of the file, therefore you will have to specify all the metadata below for Synapse to recognize it.
 
```

Comment:
If you setup your bucket for read/write you can add files to the bucket using the regular methods ...  

But if the bucket is read only or you already have content in the bucket you will have to add representations fo the files in Synapse using a script. This is done using a `FileHandle` which is a Synapse representation of the file.